cmd/account/governance.go

```diff
@@ -0,0 +1,193 @@
+package account
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strconv"
+
+	"github.com/spf13/cobra"
+
+	governance "github.com/oasisprotocol/oasis-core/go/governance/api"
+	upgrade "github.com/oasisprotocol/oasis-core/go/upgrade/api"
+	"github.com/oasisprotocol/oasis-sdk/client-sdk/go/connection"
+
+	"github.com/oasisprotocol/cli/cmd/common"
+	cliConfig "github.com/oasisprotocol/cli/config"
+)
+
+var (
+	govCmd = &cobra.Command{
+		Use:     "governance",
+		Short:   "Governance operations",
+		Aliases: []string{"gov"},
+	}
+
+	govSubmitProposalCmd = &cobra.Command{
+		Use:   "submit-proposal",
+		Short: "Submit a governance proposal",
+	}
+
+	govSubmitProposalUpgradeCmd = &cobra.Command{
+		Use:   "upgrade <descriptor.json>",
+		Short: "Submit an upgrade descriptor governance proposal",
+		Args:  cobra.ExactArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			cfg := cliConfig.Global()
+			npa := common.GetNPASelection(cfg)
+			txCfg := common.GetTransactionConfig()
+			filename := args[0]
+
+			if npa.Account == nil {
+				cobra.CheckErr("no accounts configured in your wallet")
+			}
+
+			// When not in offline mode, connect to the given network endpoint.
+			ctx := context.Background()
+			var conn connection.Connection
+			if !txCfg.Offline {
+				var err error
+				conn, err = connection.Connect(ctx, npa.Network)
+				cobra.CheckErr(err)
+			}
+
+			// Load upgrade descriptor.
+			rawDescriptor, err := os.ReadFile(filename)
+			cobra.CheckErr(err)
+
+			// Parse upgrade descriptor.
+			var descriptor upgrade.Descriptor
+			if err = json.Unmarshal(rawDescriptor, &descriptor); err != nil {
+				cobra.CheckErr(fmt.Errorf("malformed upgrade descriptor: %w", err))
+			}
+			if err = descriptor.ValidateBasic(); err != nil {
+				cobra.CheckErr(fmt.Errorf("invalid upgrade descriptor: %w", err))
+			}
+
+			// Prepare transaction.
+			tx := governance.NewSubmitProposalTx(0, nil, &governance.ProposalContent{
+				Upgrade: &governance.UpgradeProposal{
+					Descriptor: descriptor,
+				},
+			})
+
+			acc := common.LoadAccount(cfg, npa.AccountName)
+			sigTx, err := common.SignConsensusTransaction(ctx, npa, acc, conn, tx)
+			cobra.CheckErr(err)
+
+			common.BroadcastOrExportTransaction(ctx, npa.ParaTime, conn, sigTx, nil, nil)
+		},
+	}
+
+	govSubmitProposalCancelUpgradeCmd = &cobra.Command{
+		Use:   "cancel-upgrade <proposal-id>",
```

Comment:
IMO we should move this to a level above, so something like this:

```
oasis account governance submit upgrade
oasis account governance cancel upgrade
```

Also, do we need to call it cancel upgrade proposal or could `cancel <proposal-id>` be universal for any governance proposal?

Reply:
No, it is the other way around, these are different governance proposal types, currently:
* upgrade
* cancel upgrade

We may add others in the future. And you cannot cancel an arbitrary proposal, this is specifically for cancelling already scheduled upgrades.